internal/infraenv/infraenv.go

```diff
@@ -31,6 +34,35 @@ func NewManager(log logrus.FieldLogger, db *gorm.DB, objectHandler s3wrapper.API
 	}
 }
 
+func (m Manager) DeleteOrphanInfraEnvs(ctx context.Context, inactiveSince strfmt.DateTime) error {
+	log := logutil.FromContext(ctx, m.log)
+	var infraEnvs []*models.InfraEnv
+
+	if err := m.db.Where("updated_at < ?", inactiveSince).Find(&infraEnvs).Error; err != nil {
```

Comment:
Consider get with limit. we don't want GC to consume a lot of memory.

Reply:
Done